<?xml version="1.0"?>
<launch> 

  <arg name="rviz_gui"                        default="true" />
  <arg name="verbosity_level"                 default="1" />
  <arg name="optimization_backend"            default="g2o" />
  <arg name="publish_map_transform"           default="false" />
  <arg name="input_cloud_topic"               default="spinning_lidar/assembled_cloud" />


  <node name="icpslam" pkg="icpslam" type="icpslam_node" output="screen" >
    <rosparam command="load"                  file="$(find icpslam)/config/icpslam.yaml"/>
    <param name="verbosity_level"             value="$(arg verbosity_level)" />
    <!-- Available optimization backends: gtsam, g2o -->
    <param name="optimization_backend"        value="$(arg optimization_backend)" />
    <param name="publish_map_transform"       value="$(arg publish_map_transform)" />

    <remap from="/laser/point_cloud"          to="$(arg input_cloud_topic)" />
  </node>

  <!-- Rviz -->
  <group if="$(arg rviz_gui)">
    <node name="rviz_icpslam" pkg="rviz" type="rviz" args="-d $(find icpslam)/rviz/live_slam.rviz" />
  </group>
</launch>